### PR TITLE
Chemistry Update: NamePendingChem (Part 1/?)

### PR DESCRIPTION
**Pull request by Kierany9**
So after two or three months I finally got around to making that chemistry overhaul I said I would do. Or at least the first part of it. This update doesn't add anything major but does add some new chemicals as well as change some old ones.

Adds support for maximum reaction temperatures for chemicals that won't react unless cooled. Currently used by Sodium Hydride and Blindness Toxin.

**Acids:**
Sulphuric Acid: Increased melting power (10 -> 15)
Fluorosulphuric Acid: Increased melting power (20 -> 30)
Sodium Hydroxide: Replaces Lye. Is mildly corrosive (10) and has a very small chance of causing spontaneous combustion. 1 part Sodium, 1 part Oxygen, 1 part Hydrogen heated to 450k.
Sodium Hydride: Extremely corrosive (20), can cause combustion and blood loss. Made by cooling down 2 parts Carbon and 1 part Sodium Hydroxide to 175k

**Sedatives:**
Sleep Toxin: Made with four parts Sugar, one part Chloral Hydrate, this chemical allows for quick and harmless sedation. Replaces Morphine in the sleepers, and is slightly faster at sending to sleep.
Chloral Hydrate: Chloral Hydrate is slightly less poisonous than before to prevent the retardation that is going from fully healthy to dead in 10 ticks with high doses but still as effective at sending to sleep.
Midazolam: Made with 3 parts Chloral Hydrate, 1 part Fluorosulphuric Acid and 1 part Sodium Hydride (resulting in 1 part Midazolam), Midazolam is far quicker and far more toxic than Chloral Hydrate. However, it metabolizes faster than Chloral.

**Painkillers:**
Morphine: No longer sends you to sleep unless you overdose and allows you to ignore slowdown for a short amount of time. Made with Sleep Toxin and Nitrogen or by grinding poppies.
Oxycodone: A far more effective version of Morphine made with Morphine and Sodium Hydroxide, using 5u of Plasma as a catalyst.
Heroin: Allows you to ignore all pain-induced slowdowns and grants you infinite stamina as long as it's in your body. However, stuns are twice as effective against you. Made with 2 parts Oxycodone 1 part Cyanide and heated to 450k or by grinding poppies with a potency above 75.

**Stimulants:**
Ephedrine: No longer regenerates stamina. Stun resistance only works above 60 HP whereas the speed boost always works.
Hyperzine: Increases movement speed a lot but quickly drains stamina when under 60 HP. Made with sugar, phosphorous and sulphuric acid. Uses 5u of Plasma as a catalyst.
Methamphetamine: All previous nerfs to methamphetamine have been undone. Stun resistance is now 50% instead of 66% and constantly drains your stamina. Mixing with heroin is its own punishment. Made with 2 parts Ephedrine, 2 parts Hyperzine, 1 part Iodine

**Misc New Chemicals:**
Amperium: Immediately causes an arc flash, electrocuting people and possibly igniting and causing EMPs in higher amounts, unless it is stabilized. Once stabilized, it still has a chance of shocking anybody nearby. Made with Flash Powder, Teslium and Hyperzine.
Ignis: At random intervals, causes you to spontaneously combust if not on fire and doubles the intensity of any fire already burning you. Made with Sodium Hydride, Napalm and Oxygen.
Blindness Toxin: Makes you blind for a short period of time.1 Uranium, 2 Carbon, 2 Water, cooled to under 450k
Cryoxadin(Cryotoxin): Does extreme damage in the cold and is easily confusable for Cryoxadone in name and appearance.

**Misc Changes:**
Water+Potassium: Uses different explosion size scaling. Amounts below 200u will be much more effective than before, but loses effectiveness quickly above 200u.
Miner's Salve Recipe: Removed one of the two recipes.
Mute Toxin: Now has a less uranium-intensive recipe (1 Uranium, 2 Carbon, 2 Water heated to 450k)

### CODE REVIEW

**Comment by Voratius** [comment]
-1/3rd
no ty

**Comment by TDcoolguy300** [comment]
Speaker confirmed to be salty about Morphine changes.

**Comment by blobbernaut** [comment]
Ephedrine: No longer regenerates stamina or makes you run quickly and only works above 60 HP

only bad one tbh

**Comment by blobbernaut** [comment]
also buff epinephrine, it's literally adrenaline why is it so useless

**Comment by Kierany9** [comment]
I'll be touching epinephrine (and atropine) with the rest of the medicines later on. I've got enough stuff in one PR as it is...

brb adding heroin to poppies

File: code/modules/reagents/Chemistry-Reagents/Drug-Reagents.dm [hunk 1]
```diff
@@ -162,46 +162,30 @@
 /datum/reagent/drug/methamphetamine
 	name = "Methamphetamine"
 	id = "methamphetamine"
-	description = "Reduces stun times by about 300%, speeds the user up, and allows the user to quickly recover stamina while dealing a small amount of Brain damage. If overdosed the subject will move randomly, laugh randomly, drop items and suffer from Toxin and Brain damage. If addicted the subject will constantly jitter and drool, before becoming dizzy and losing motor control and eventually suffer heavy toxin damage."
+	description = "Halves the duration of stuns and greatly increases movement speed. However, it drains an ever-increasing ammount of stamina over time and being stunned will force the body to metabolize the drug faster."
```
Comment (McBawbaggings) [inline code/modules/reagents/Chemistry-Reagents/Drug-Reagents.dm]: "Ammount"

File: code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm [hunk 2]
```diff
@@ -813,7 +678,176 @@
...
+		if(I)
+			M.drop_item()
+			M.visible_message("<span class='danger'>[M] shudders and drops [I] on the floor!</span>")
+			M.Jitter(2)
+	return
+
+/datum/reagent/medicine/oxycodone
+	name = "Oxycodone"
+	id = "oxycodone"
+	description = "A strong painkiller that allows the user to move at full speed until in critical condition for a decent ammount of time. May send to sleep or cause drowsiness once it wears off."
```
Comment (McBawbaggings) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: "Ammount"

**Comment by blobbernaut** [comment]
I always thought it would be neat if there were mechanics for players to create small amounts of adrenaline on their own. Seeing a horror form ling emerge, hearing gunshots, that sort of thing.

**Comment by Voratius** [comment]
Blobber, except when I suggested that people said it was a stupid idea.

**Comment by GeneralGoose** [comment]
so you basically added stronger versions of chems that are aquireable through such simple reactions
its cool but i dont really like it

also if you're changing chems, you should fiks miner's salve.
on ingestion it weakens and stuns you. but there is no limit to it, so if you are in a cloud of miner's salve you're perma stunned. also works with spray bottle
and even though it's not that easy to make now it's still cancerous as even 1u is enough...
https://github.com/Kierany9/HippieStation13/blob/f56b7b0e7ad26a2ea057a3a62cd4c6b0e390edf5/code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm#L269

File: code/modules/reagents/Chemistry-Reagents/Drug-Reagents.dm [hunk 3]
```diff
@@ -162,46 +162,33 @@
...
+	M.AdjustWeakened(-1)
+	M.adjustStaminaLoss(4) // Is actually -2 per tick, humans regenerate 2 per tick
+	M.adjustToxLoss(0.4*REM)
+	M.status_flags |= GOTTAGOREALLYFAST
+	M.sleeping = max(0,M.sleeping - 3)
 	if(prob(5))
 		M.emote(pick("twitch", "shiver"))
+	M.Jitter(2)
+	if(M.stunned || M.weakened)
+		holder.remove_reagent("methamphetamine", 0.75 * REAGENTS_METABOLISM)
```
Comment (JohnGinnane) [inline code/modules/reagents/Chemistry-Reagents/Drug-Reagents.dm]: Why not use metabolization_rate as defined for this drug? It means you can also remove the snowflakey code below for heroin (Set its metabolization_rate to 2 * REAGENTS_METABOLISM)

File: code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm [hunk 4]
```diff
@@ -813,7 +678,176 @@
...
+	id = "sleeptoxin"
+	description = "A safe way to send a patient to sleep. 8 units is required to send them to sleep."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = REAGENTS_METABOLISM
+
+/datum/reagent/medicine/sleeptoxin/on_mob_life(mob/living/carbon/human/M)
+	if(current_cycle == 7)
+		M << "<span class='warning'>You start to feel tired...</span>" //Warning when the victim is starting to pass out
+	if(current_cycle > 15)
```
Comment (JohnGinnane) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: Description says 8 units is required but does this mean after 15 units they fall asleep?

File: code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm [hunk 5]
```diff
@@ -813,7 +678,176 @@
...
+	reagent_state = LIQUID
+	color = "#E6E6BE"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	overdose_threshold = 30
+
+/datum/reagent/medicine/hyperzine/on_mob_life(mob/living/M)
+	if(M.health > 80)
+		M.status_flags |= GOTTAGOREALLYFAST
+	else if(M.health > 60)
+		M.status_flags |= GOTTAGOFAST
```
Comment (JohnGinnane) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: Do these flags get removed later on?

**Comment by Kierany9** [comment]
>also if you're changing chems, you should fiks miner's salve.

I removed the salve recipe that doesn't require plasma so that it's only viable for miners to make instead of being used as an instant stun that's cheaper to make than neurotoxin(the drink).

>also what was even the point of this 'overhaul'? what did you want to achieve

Add cool new shit to goofchem, try and balance things out a little and make medicines actually worth making. Not doing everything in one PR because there's simply so much to do.

File: code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm (hunk 5 above)
Comment (Kierany9) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: Nope, they should.

File: code/modules/reagents/Chemistry-Reagents/Pyrotechnic-Reagents.dm [hunk 6]
```diff
@@ -200,3 +200,50 @@
...
+//NamePendingChem Chems
+
+/datum/reagent/amperium
+	name = "Amperium"
+	id = "amperium"
+	description = "A reaction used to cause arc flashes. Is somewhat unstable even if stabilized and will decay over time, launching lightning bolts at anybody unlucky enough to be close to it."
+	color = "#BCA1E6"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+
+/datum/reagent/amperium/on_mob_life(mob/living/M)
```
Comment (JohnGinnane) [inline code/modules/reagents/Chemistry-Reagents/Pyrotechnic-Reagents.dm]: Does drinking this eletrocute the holder?

File: code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm (hunk 4 above)
Comment (Kierany9) [inline code/modules/reagents/Chemistry-Reagents/Medicine-Reagents.dm]: It means after 15 ticks they fall asleep. The default metabolization rate (REAGENTS_METABOLISM) is 0.4 units per tick. Doing some math, it'll send people to sleep in...6 units....

fixing

File: code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm [hunk 7]
```diff
@@ -666,16 +686,104 @@
...
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if(H.vessel)
+			H.vessel.remove_reagent("blood",rand(1, 5)) //Drain blood with various effectiveness
+	..()
+
+/datum/reagent/toxin/bleach
+	name = "Bleach"
+	id = "bleach"
+	description = "A powerful cleaner.Toxic if injested"
```
Comment (JohnGinnane) [inline code/modules/reagents/Chemistry-Reagents/Toxin-Reagents.dm]: Should read "A powerful cleaner. Toxic if ingested."

File: code/modules/reagents/Chemistry-Reagents/Pyrotechnic-Reagents.dm (hunk 6 above)
Comment (Kierany9) [inline code/modules/reagents/Chemistry-Reagents/Pyrotechnic-Reagents.dm]: Yes, unless wearing gloves.

File: code/modules/reagents/Chemistry-Reagents/Drug-Reagents.dm (hunk 3 above)
Comment (Kierany9) [inline code/modules/reagents/Chemistry-Reagents/Drug-Reagents.dm]: Done

**Comment by trifornt-tinfornt** [comment]
Adding stronger chloral, which takes two seconds to make?
_g e n i u s_
The meth nerf is also pretty much just removing the nerfs you added to meth to begin with and then adding them back but slightly different. I already suggested that meth and anti-stun chems should be able to work (albeit not as efficiently as they do right now) but make combat worthless as you can pretty much do nothing while it's running in your system.

Besides a couple chem changes and additions (like the morphine change, like the miner's salve change, like that mediborg is now useful. That's honestly about it) I think this honestly won't hold that much merit. And some of it is pretty dumb, like "stronger chloral".
You know that one chem that can knock you out for about seven minutes when you are shot with 15u of it and can't be fixed by simply drinking coffee and sugar like it could before-rebase?
Let's add a more powerful version of that which ALSO can't be stopped by those two methods!
5u neurotoxin 5u perfluorodecalin gg win shot

**Comment by Kierany9** [comment]
> 5u neurotoxin 5u perfluorodecalin 5u midazolam gg win shot 

Except for the part where it metabolizes so fast you'll be asleep for a grand total of one tick?

As for Meth, I don't think that it should be worthless in combat but it definitely needs an actual counter. Despite that, right now, the "increased metabolism while stunned" is actually a pretty big deal since if you get stunlocked(which is easier due to the reduction in stun resistance) the meth will quickly run out and leave them stranded in a worse state than before.

**Comment by Kierany9** [comment]
Midazolam recipe made more complicated due to popular demand.

Also reduced fluorosulphuric acid's power to 30 and sodium hydride's to 20 to prevent the bullshit that was poly smoke from happening again.

**Comment by trifornt-tinfornt** [comment]
Make it 35.

**Comment by Kierany9** [comment]
will redo/reopen later

**Pull request closed without merging**